### ISSUE TITLE
AutoStream - Use a stream instead of autoplaylist.txt

### ISSUE THREAD

**Issue by SeasonalFerret** [Just-Some-Bots/MusicBot#1587]
Please tick all applicable boxes.

- [x] I am using Python 3.5 or higher (run `python --version` on the command line)
- [x] I have followed the [official guides](https://github.com/Just-Some-Bots/MusicBot/wiki) to install the bot for my system
- [x] I have updated my dependencies to the latest version using the appropriate `update` script

Which version are you using?

- [x] The latest `master` version (1.9.7)
- [ ] The latest `review` version

What type of issue are you creating?

- [ ] Bug
- [x] Feature request
- [ ] Question

----

### Feature Request

Would like to have MusicBot play from a stream when no music is in the queue. When a song is requested, close the stream and play the song immediately. Once the queue is done, attempt to re-connect to the stream.

Examples of features could be:

1. An "autostream.txt" (compared to autoplaylist.txt), which contains a stream URL for the bot to connect to (similar to !stream {url}).
2. !autostream {url} [save], would select a URL for autostream to use for that session. If implemented alongside "autostream.txt", adding "save" to the end of the command would override the URL used in autostream.txt
3. Typing "!autostream" could toggle between using a stream or autoplaylist.txt when not used alongside a command. That, or "!autostream {on/off}.
4. A setting in options.ini to always use autostream instead of autoplaylist.txt on startup. When disabled, MusicBot will always play from autoplaylist.txt instead of autostream.txt unless !autostream is enabled or a stream URL has been set.

Example !help: 

```
Usage:
   !autostream {on/off/song_link} [save]

Automatically enqueues a media stream when no songs are playing.
Replaces autoplaylist in favor of automatically playing a stream when no songs are in the queue. See {}stream for more info.

**Comment by SeasonalFerret** [comment]
I'm really proud of this suggestion. Being able to load from stream URL's instead of autoplaylist seems like a wonderful idea.

!stream is compatible with my favorite online FM radio, so I would like it to always play their stream. But I also would like users to join and play their own music, especially when I'm offline.
This command would allow me to use a preset stream URL that would automatically play when no other song is being used, similar to how autoplaylist.txt works. The only exception is that when a user requests a song, the stream is immediately closed and the bot plays their requested song.

**Comment by NeddyPlays** [comment]
+1 would love this. I can't set the radio stream to auto play at the moment.

**Comment by btkostner** [comment]
It seems you can have it stream twitch by putting it in the autoplaylist file, but because it recognises it as a track it will play forever until you use `!skip`.

**Comment by SeasonalFerret** [comment]
Considering the issue with #1947, I'd imagine this would help? Instead of playing through the autoplaylist, loading video after video, the user could use autostream to buffer some radio music until youtube stops being dumb.

**Comment by TheerapakG** [comment]
@SylveonBottle or just use other services.

On another note, this is kinda implemented in #1973 (it have autostream but it does not work like how this request described). It also allows queueing local entries so you do not have to worry about getting banned from these services at all.
